oidc-exchange.py

```diff
@@ -0,0 +1,157 @@
+import os
+import sys
+from pathlib import Path
+from textwrap import dedent
+from typing import NoReturn
+from urllib.parse import urlparse
+
+import id  # pylint: disable=W0622
+import requests
+
+_GITHUB_STEP_SUMMARY = Path(os.getenv("GITHUB_STEP_SUMMARY"))
+
+_TOKEN_RETRIEVAL_FAILED_MESSAGE = dedent(
+    """
+    OIDC token retrieval failed: {identity_error}
+
+    This generally indicates a workflow configuration error, such as insufficient
+    permissions. Make sure that your workflow has `id-token: write` configured
+    at either the workflow or job level, e.g.:
+
+    ```yaml
+    permissions:
+      id-token: write
+    ```
+    """,
+)
```

Comment:
@woodruffw could you explain why this can't be like this? I don't understand...
````suggestion
_TOKEN_RETRIEVAL_FAILED_MESSAGE = """
OIDC token retrieval failed: {identity_error}

This generally indicates a workflow configuration error, such as insufficient
permissions. Make sure that your workflow has `id-token: write` configured
at either the workflow or job level, e.g.:

```yaml
permissions:
  id-token: write
```
"""
````

Reply:
No reason, it can!